src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+        """
+        with photons_file.open("r", encoding="utf-8") as fh:
+            for line in fh:
+                if not line.strip() or line.lstrip().startswith("#"):
+                    continue
+                yield line.split()
+
+    def _append_values(
+        self,
+        parts,
+        col_idx,
+        focal,
+        primary,
+        secondary,
+        radius_m,
+        secondary_radius_m,
+        primary_hit_x_m,
+        primary_hit_y_m,
+        secondary_hit_x_m,
+        secondary_hit_y_m,
+    ):
+        """Append parsed values from parts into target arrays if valid.
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Column indices used to read values.
+        focal, primary, secondary : list | None
+            Output arrays to append into.
+        radius_m, secondary_radius_m : list | None
+            Output arrays for radii in meters.
+        primary_hit_x_m, primary_hit_y_m, secondary_hit_x_m, secondary_hit_y_m : list | None
+            Output arrays for hit coordinates in meters.
+        """
+        foc_ok, foc_val = self._parse_float(parts, col_idx.get("focal"))
+        if not foc_ok:
+            return
+        focal.append(foc_val)
+        if not self.calculate_primary_secondary_angles:
+            return
+
+        self._append_primary_secondary_angles(parts, col_idx, primary, secondary)
+        self._append_primary_hit_geometry(
+            parts, col_idx, radius_m, primary_hit_x_m, primary_hit_y_m
+        )
+        self._append_secondary_hit_geometry(
+            parts, col_idx, secondary_radius_m, secondary_hit_x_m, secondary_hit_y_m
+        )
+
+    def _append_primary_secondary_angles(self, parts, col_idx, primary, secondary):
+        """Append primary/secondary angle values (or NaN) if arrays are provided.
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Indices for angle columns.
+        primary, secondary : list | None
+            Output arrays to append into.
+        """
+        if primary is not None:
+            primary.append(self._parse_float_with_nan(parts, col_idx.get("primary")))
+        if secondary is not None:
+            secondary.append(self._parse_float_with_nan(parts, col_idx.get("secondary")))
+
+    def _append_primary_hit_geometry(
+        self, parts, col_idx, radius_m, primary_hit_x_m, primary_hit_y_m
+    ):
+        """Append primary-mirror hit geometry (radius and x/y in meters).
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Indices for hit coordinate columns.
+        radius_m, primary_hit_x_m, primary_hit_y_m : list | None
+            Output arrays to append into.
+        """
+        x_ok, x_cm = self._parse_float(parts, col_idx.get("prim_x"))
```

Comment:
I think the following would be easier to read:

```
    x_ok, x_cm = self._parse_float(parts, col_idx.get("prim_x"))
    y_ok, y_cm = self._parse_float(parts, col_idx.get("prim_y"))

    if x_ok and y_ok:
        x_m, y_m = x_cm / 100.0, y_cm / 100.0
        r_m = (x_cm**2 + y_cm**2) ** 0.5 / 100.0
    else:
        x_m = y_m = r_m = float("nan")

    if radius_m is not None:
        radius_m.append(r_m)
    if primary_hit_x_m is not None:
        primary_hit_x_m.append(x_m)
    if primary_hit_y_m is not None:
        primary_hit_y_m.append(y_m)
```

Reply:
fixed